--- origsrc/libsocialweb-0.25.17/libsocialweb/sw-core.c	2011-02-22 08:03:00.000000000 -0600
+++ src/libsocialweb-0.25.17/libsocialweb/sw-core.c	2011-05-18 02:04:25.467751000 -0500
@@ -204,7 +204,7 @@ load_modules_from_dir (SwCore *core)
 
   while ((fi = g_file_enumerator_next_file (enumerator, NULL, &error)) != NULL)
   {
-    if (g_str_has_suffix (g_file_info_get_name (fi), ".so"))
+    if (g_str_has_suffix (g_file_info_get_name (fi), "." G_MODULE_SUFFIX))
     {
       char *module_path;
 
@@ -278,15 +278,13 @@ load_modules_from_string (SwCore *core,
   modules = g_strsplit (s, ",", 0);
 
   for (basename = modules; *basename; basename++) {
-    char *name, *path;
+    char *path;
 
-    name = g_strconcat ("lib", *basename, NULL);
-    path = g_build_filename (SOCIALWEB_SERVICES_MODULES_DIR, name, NULL);
+    path = g_module_build_path (SOCIALWEB_SERVICES_MODULES_DIR, *basename);
 
     load_module (core, path);
 
     g_free (path);
-    g_free (name);
   }
 
   g_strfreev (modules);
